pkg/kds/v2/reconcile/reconciler.go

```diff
@@ -64,19 +65,37 @@ func (r *reconciler) Clear(ctx context.Context, node *envoy_core.Node) error {
 	return nil
 }
 
-func (r *reconciler) Reconcile(ctx context.Context, node *envoy_core.Node) error {
-	new, err := r.generator.GenerateSnapshot(ctx, node)
+func (r *reconciler) Reconcile(ctx context.Context, node *envoy_core.Node, changedTypes map[core_model.ResourceType]struct{}) error {
+	id, err := r.hashId(ctx, node)
 	if err != nil {
 		return err
 	}
-	if new == nil {
-		return errors.New("nil snapshot")
+	old, _ := r.cache.GetSnapshot(id)
+
+	// construct builder with unchanged types from the old snapshot
+	builder := cache_v2.NewSnapshotBuilder()
+	if old != nil {
+		for _, typ := range util_kds_v2.GetSupportedTypes() {
```

Comment:
`GetSupportTypes` returns all types we sync (both Zone->Global and Global->Zone), but we build snapshots with fewer types (i.e. `reg.ObjectTypes(model.HasKDSFlag(model.ProvidedByGlobal))`). This probably won't result in a bug because you're checking `len(oldRes)` but I'd fix it to avoid potential bugs in the future

Reply:
Yeah, this is intentional because like you said, we just won't set those resources. We do the same with in version function

Reply:
I'm merging as is. If you feel this is important, please create an issue. Thx!